### ISSUE TITLE
Form inside a <BsModalSimple> are not submitted on enter

### ISSUE THREAD

**Issue by jelhan** [kaliber5/ember-bootstrap#918]
A form is submitted on enter only

- if it has at least one submit button ( `<button type="submitt">` or `<input type="submit">`) or
- if it has only one input.

A form in `<BsModalSimple>` does not have a submit button. The modal button works as an submit button but isn't one technically. It has `type="button"` and is not a child element of `<form>`.

To fix this the `type` attribute of the button must be changed to `"submit"` and it must be linked to the form by setting `form` attribute to the ID of the form.

This is not an easy one with existing API cause currently the `<BsModalSimple>` is not aware if it's used with a form or not. Only if submit button is clicked, it looks if it has a `<form>` as a child element and triggers a submit event on it: https://github.com/kaliber5/ember-bootstrap/blob/dae98866a964b6d4c32f1dbf71879ed500c9fde5/addon/components/base/bs-modal.js#L431-L445

Adding a event listener for enter key might be a quick solution but doesn't feel like a fix of the underlying issues. The existing API is also an issue for handling of interim states for form's inside a modal (e.g. showing loading state on that button). See #638 for details. It might be the right time to fix the underlying issues with the current API.

I'm not an accessibility expert but I think that's also an a11y issue.

**Comment by simonihmig** [comment]
Just stumbled upon the fact the we have a `submitOnEnter` property on the form! That would probably solve that use case?

Maybe we should yield the `BsForm` from the modal, so this is auto-set?

**Comment by jelhan** [comment]
> Just stumbled upon the fact the we have a `submitOnEnter` property on the form! That would probably solve that use case?

Yeah. Wasn't aware of that argument to be honest. Lets treat this one as a documentation issue. I think `@submitOnEnter` should be set in the example given in API docs of `<BsModalSimple>`.

> Maybe we should yield the `BsForm` from the modal, so this is auto-set?

I fear that more changes are needed to solve all the issues with using a form inside a modal. E.g. the loading indication is a related one. I'm tending towards introducing a `<BsModalForm>` component that supports that use case better. Will try to propose an API for that one in another issue.
